include/crm/common/results.h

```diff
@@ -42,7 +42,6 @@ extern "C" {
 #  define CRM_ASSERT(expr) do {                                              \
         if (!(expr)) {                                                       \
             crm_abort(__FILE__, __func__, __LINE__, #expr, TRUE, FALSE);     \
-            abort(); /* crm_abort() doesn't always abort! */                 \
```

Comment:
It's a lot noisier but spot-checking a few `CRM_ASSERT()` locations, coverity doesn't seem to complain about them.

Reply:
> It's a lot noisier but spot-checking a few `CRM_ASSERT()` locations, coverity doesn't seem to complain about them.

Even without any previously waived issues? It must have improved in that area

> cppcheck: error: unrecognized command line option: "-pthread".

I wonder if that's in GLIB_CFLAGS

Reply:
> -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-6 -pthread -D__GNUC__

yep. maybe we should just pull out -l and -D options from GLIB_FLAGS.

Reply:
> Even without any previously waived issues? It must have improved in that area

Unless those are automatically hidden when running locally... which I doubt, because there were several issues that did show up, and they all looked unlikely to be in code this PR touches.

> yep. maybe we should just pull out -I and -D options from GLIB_FLAGS.

Probably a good plan.

Reply:
`cppcheck` doesn't mind the updated `crm_abort()` either.

Reply:
I just went through and corrected as many cppcheck errors as I could. These are the ones that remain, and they're false positives (despite cppcheck's documentation saying it's super conservative and aims for zero false positives lol):
```
^[[1m../lib/cib/cib_ops.c:549:5: ^[[31merror:^[[39m There is an unknown macro here somewhere. Configuration is required. If PCMK_XA_ID is a macro then please configure it. [unknownMacro]^[[0m
    crm_trace("Processing update for <%s%s%s%s>", object_name,
    ^
^[[1m../daemons/pacemakerd/pcmkd_subdaemons.c:485:9: ^[[31merror:^[[39m There is an unknown macro here somewhere. Configuration is required. If VALGRIND_BIN is a macro then please configure it. [unknownMacro]^[[0m
        crm_info("Forked child %lld for process %s%s",
        ^
^[[1m../lib/common/output.c:135:24: ^[[31merror:^[[39m Null pointer dereference: name [ctunullpointer]^[[0m
    name_copy = strdup(name);
                       ^
^[[1m../lib/common/tests/output/pcmk__register_format_test.c:29:5: ^[[2mnote:^[[0m Calling function pcmk__register_format, 2nd argument is null
    pcmk__assert_asserts(pcmk__register_format(NULL, NULL, null_create_fn, NULL));
    ^
^[[1m../lib/common/output.c:135:24: ^[[2mnote:^[[0m Dereferencing argument name that is null
    name_copy = strdup(name);
                       ^
^[[1m../lib/common/strings.c:911:6: ^[[31merror:^[[39m Null pointer dereference: end [ctunullpointer]^[[0m
    *end = PCMK__PARSE_INT_DEFAULT;
     ^
^[[1m../lib/common/tests/strings/pcmk__parse_ll_range_test.c:30:5: ^[[2mnote:^[[0m Calling function pcmk__parse_ll_range, 3rd argument is null
    pcmk__assert_asserts(pcmk__parse_ll_range("1234", &start, NULL));
    ^
^[[1m../lib/common/strings.c:911:6: ^[[2mnote:^[[0m Dereferencing argument end that is null
    *end = PCMK__PARSE_INT_DEFAULT;
     ^
```

I don't know what to do about the first two. `$(top_srcdir)/include` is the first include directive passed to `cppcheck`, so it should get every macro definition. Also, it doesn't see a problem with those two macros anywhere else they're used.

The last six are in three groups of two. In both cases, a `CRM_ASSERT()` should catch a null pointer[*].

I went back and re-added the `abort()` to `CRM_ASSERT()`, and the same errors persist. So re-adding it won't fix the "problem"; might as well drop it.

[*] Actually, in the `output.c` one, there is a legitimate null-dereference on main. But the same cppcheck errors persist even after I fix those.